src/validation.cpp

```diff
@@ -812,7 +812,7 @@ bool MemPoolAccept::PreChecks(ATMPArgs& args, Workspace& ws)
         return state.Invalid(TxValidationResult::TX_INPUTS_NOT_STANDARD, "bad-txns-nonstandard-inputs");
     }
 
-    if (DatacarrierBytes(tx, m_view) > m_pool.m_max_datacarrier_bytes) {
+    if (m_pool.m_max_datacarrier_bytes && DatacarrierBytes(tx, m_view) > *m_pool.m_max_datacarrier_bytes) {
```

Comment:
A suggestion: Maybe this code change could solve the issue when max_datacarrier_byte is "0" bytes:

if (!m_pool.m_max_datacarrier_bytes || (DatacarrierBytes(tx, m_view) > *m_pool.m_max_datacarrier_bytes))